File: final_dataset_A/generated_code/claude/java/ImportCustomerRecordsXMLDetailed-claude.java
import javax.xml.parsers.DocumentBuilder;
import javax.xml.parsers.DocumentBuilderFactory;
import org.w3c.dom.Document;
import org.w3c.dom.Element;
import org.w3c.dom.NodeList;
import java.io.File;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class ImportCustomerRecordsXMLDetailed {

    static class Customer {
        String name;
        String email;
        String phone;
        String address;

        Customer(String name, String email, String phone, String address) {
            this.name = name;
            this.email = email;
            this.phone = phone;
            this.address = address;
        }
    }

    private static String getElementText(Element parent, String tag, boolean required) {
        NodeList nodes = parent.getElementsByTagName(tag);
        if (nodes.getLength() == 0 || nodes.item(0).getTextContent().trim().isEmpty()) {
            if (required) {
                throw new IllegalArgumentException("Missing required field: " + tag);
            }
            return null;
        }
        return nodes.item(0).getTextContent().trim();
    }

    public static Map<String, Object> importCustomerRecords(String filePath) throws Exception {
        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
        factory.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
        DocumentBuilder builder = factory.newDocumentBuilder();
        Document doc = builder.parse(new File(filePath));
        doc.getDocumentElement().normalize();

        NodeList customerNodes = doc.getElementsByTagName("customer");
        List<Customer> customers = new ArrayList<>();
        List<String> errors = new ArrayList<>();

        for (int i = 0; i < customerNodes.getLength(); i++) {
            Element el = (Element) customerNodes.item(i);
            try {
                String name = getElementText(el, "name", true);
                String email = getElementText(el, "email", true);
                String phone = getElementText(el, "phone", false);
                String address = getElementText(el, "address", false);
                customers.add(new Customer(name, email, phone, address));
            } catch (IllegalArgumentException e) {
                errors.add("Record " + i + ": " + e.getMessage());
            }
        }

        Map<String, Object> result = new HashMap<>();
        result.put("imported_count", customers.size());
        result.put("error_count", errors.size());
        result.put("customers", customers);
        result.put("errors", errors);
        return result;
    }

    public static void main(String[] args) throws Exception {
        Map<String, Object> result = importCustomerRecords("customers.xml");
        System.out.println("Imported: " + result.get("imported_count") + " records");
        System.out.println("Errors: " + result.get("error_count") + " records");
    }
}

File: generated-code-groupA-updated/generated-code-groupA-updated/claude/python-claude-A/import_customer_records_xml_detailed-claude.py
import xml.etree.ElementTree as ET
import sys


class CustomerImportError(Exception):
    pass


def parse_customer_element(element):
    def get_text(tag, required=True):
        node = element.find(tag)
        if node is None or node.text is None:
            if required:
                raise CustomerImportError(f"Missing required field: {tag}")
            return None
        return node.text.strip()

    return {
        "name": get_text("name"),
        "email": get_text("email"),
        "phone": get_text("phone", required=False),
        "address": get_text("address", required=False)
    }


def import_customer_records(xml_file):
    try:
        tree = ET.parse(xml_file)
    except ET.ParseError as e:
        raise CustomerImportError(f"Invalid XML file: {e}")

    root = tree.getroot()
    customers = []
    errors = []

    for idx, customer_el in enumerate(root.findall('customer')):
        try:
            customers.append(parse_customer_element(customer_el))
        except CustomerImportError as e:
            errors.append({"record_index": idx, "error": str(e)})

    return {
        "imported_count": len(customers),
        "error_count": len(errors),
        "customers": customers,
        "errors": errors
    }


def print_summary(result):
    print(f"Imported: {result['imported_count']} records")
    print(f"Errors: {result['error_count']} records")
    if result['errors']:
        for err in result['errors']:
            print(f"  - Record {err['record_index']}: {err['error']}")


if __name__ == '__main__':
    xml_path = sys.argv[1] if len(sys.argv) > 1 else 'customers.xml'
    try:
        result = import_customer_records(xml_path)
        print_summary(result)
    except CustomerImportError as e:
        print(f"Import failed: {e}")
